src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx

```diff
@@ -45,54 +52,41 @@ export default class SentryAppInstallations extends AsyncView {
   };
 
   install = app => {
-    const {orgId} = this.props;
-
-    const success = install => {
-      this.redirectUser({install: {...install}, app: {...app}});
-    };
-
-    const error = err => {
-      addErrorMessage(t(`Unable to install ${app.name}`));
-    };
-
-    const opts = {
-      method: 'POST',
-      data: {slug: app.slug},
-      success,
-      error,
-    };
-
-    api.request(`/organizations/${orgId}/sentry-app-installations/`, opts);
+    const {orgId, api} = this.props;
+    installSentryApp(api, orgId, app).then(
+      data => {
+        this.redirectUser({install: {...data}, app: {...app}});
+      },
+      () => {}
+    );
   };
 
   uninstall = install => {
+    const {api} = this.props;
     const origInstalls = [...this.state.installs];
     const installs = this.state.installs.filter(i => install.uuid != i.uuid);
     this.setState({installs});
 
-    const success = () => {
-      addSuccessMessage(t(`${install.app.slug} successfully uninstalled.`));
-    };
-
-    const error = err => {
-      this.setState({origInstalls});
-      addErrorMessage(t(`Unable to uninstall ${install.app.name}`));
-    };
-
-    const opts = {
-      method: 'DELETE',
-      success,
-      error,
-    };
+    uninstallSentryApp(api, install).then(
+      () => {},
+      () => {
+        this.setState({origInstalls});
+        addErrorMessage(t(`Unable to uninstall ${install.app.name}`));
+      }
+    );
```

Comment:
If you only need to set an error handler you can use `promise.catch()`. This works as well though.